lib/rspec/rabl/attribute_matcher.rb

```diff
@@ -11,12 +11,25 @@ def description
       end
 
       def failure_message
-        "expected #{expected_value.inspect} in #{attribute_path}\n  got #{rendered_value.inspect}"
+        if nil_value?
+          "nil value when testing.\n expected #{expected_value.inspect} in #{attribute_path}\n  got #{rendered_value.inspect}\n If you want to test for a nil value, please use `with_value(nil)`"
+        else
+          "expected #{expected_value.inspect} in #{attribute_path}\n  got #{rendered_value.inspect}"
+        end
       end
 
       def matches?(subject)
         @subject = subject
-        attribute_rendered? && (!expected_value_set || rendered_value == expected_value)
+        attribute_rendered? && (!expected_value_set || rendered_value == expected_value) && nil_value_check?
+      end
+
+      def nil_value?
```

Comment:
this method name is a little confusing to me. Is the intent to check for whether the rendered value is nil? Or whether the test's author intended it to be nil?

It seems like we are using it to decide whether the author intended to check for nil so maybe we should rename the method.